src/python/pants/backend/docker/util_rules/docker_build_context.py

```diff
@@ -179,8 +180,12 @@ def create(
     ) -> DockerBuildContext:
         version_context: dict[str, dict[str, str] | DockerVersionContextValue] = {}
 
-        # FROM tags for all stages.
-        for stage, tag in [tag.split(maxsplit=1) for tag in dockerfile_info.version_tags]:
+        # Go over all FROM tags and names for all stages.
+        stage_names: set[str] = set()
+        stage_tags = (tag.split(maxsplit=1) for tag in dockerfile_info.version_tags)
+        for idx, (stage, tag) in enumerate(stage_tags):
+            if stage != f"stage{idx}":
+                stage_names.add(stage)
```

Comment:
How hard would it be to add a test for this?

Reply:
Done :)

https://github.com/pantsbuild/pants/pull/13956/files#diff-1abf5f4e44f1392638e34794a929bb6cd29b2fb6d4f08913fb41d75e726804a0R549-R568